src/lib/util/Config.ts

```diff
@@ -90,6 +92,18 @@ export class Config {
 		return this.configContent.engines.find(e => e.name === name);
 	}
 
+	public async setEngineConfig(name: string, ecc: EngineConfigContent): Promise<void> {
+		// If there's an entry in the Engine array that has the same name as the one we were provided, override it with
+		// the new one.
+		const idx = this.configContent.engines.findIndex(e => e.name === name);
```

Comment:
It seems like ConfigContent is moving from an immutable to mutable object, it would be a good time to evaluate its current structure. 

Should ConfigContent#engines be a Map instead of an Array? It looks like most of the code performs a find on name. Leaving it as an Array leaves it open to bugs of multiple entries with the same engine name.